### PR TITLE
Add BLE and thread pairing capabilities to chip-tool-darwin.

### PR DESCRIPTION
**Pull request by krypton36**
#### Problem
BLE and Thread commissioning is missing in chip-tool-darwin.

#### Change overview
- Add Thread and BLE pair commissioning to chip-tool-darwin.

#### Testing
- Validated pairing and commissioning through BLE

### CODE REVIEW

File: examples/chip-tool-darwin/commands/pairing/PairingDelegateBridge.h [hunk 1]
```diff
@@ -21,6 +21,9 @@
 @interface CHIPToolPairingDelegate : NSObject <CHIPDevicePairingDelegate>
 @property PairingCommandBridge * commandBridge;
 @property chip::NodeId deviceID;
+@property CHIPDeviceController * mCommissioner;
```
Comment (bzbarsky-apple) [inline examples/chip-tool-darwin/commands/pairing/PairingDelegateBridge.h:24]:
```suggestion
@property CHIPDeviceController * commissioner;
```

File: examples/chip-tool-darwin/commands/pairing/PairingDelegateBridge.h [hunk 2]
```diff
@@ -21,6 +21,9 @@
...
+@property CHIPCommissioningParameters * mParams;
```
Comment (bzbarsky-apple) [inline examples/chip-tool-darwin/commands/pairing/PairingDelegateBridge.h:25]:
```suggestion
@property CHIPCommissioningParameters * commissioningParams;
```

File: examples/chip-tool-darwin/commands/pairing/PairingDelegateBridge.mm [hunk 3]
```diff
@@ -41,10 +42,21 @@ - (void)onStatusUpdate:(CHIPPairingStatus)status
...
+    NSError * __block commissionError;
     CHIP_ERROR err = [CHIPError errorToCHIPErrorCode:error];
+    if (err != CHIP_NO_ERROR) {
+        _commandBridge->SetCommandExitStatus(err);
+        return;
+    }
     ChipLogProgress(chipTool, "Pairing Complete: %s", chip::ErrorStr(err));
-
-    _commandBridge->SetCommandExitStatus(err);
+    [_mCommissioner commissionDevice:_deviceID
```
Comment (bzbarsky-apple) [inline examples/chip-tool-darwin/commands/pairing/PairingDelegateBridge.mm:52]: In the "ethernet" case, won't we land here _after_ commissioning is done?  Please check what happens there.

**Review by bzbarsky-apple — APPROVED** [review]

File: examples/chip-tool-darwin/commands/pairing/PairingDelegateBridge.mm (hunk 3 above)
Comment (krypton36) [reply examples/chip-tool-darwin/commands/pairing/PairingDelegateBridge.mm:52]: Ran through the ethernet scenario and onPairingComplete does not get called and is commissioned successfully.

File: examples/chip-tool-darwin/commands/pairing/PairingDelegateBridge.h (hunk 1 above)
Comment (krypton36) [reply examples/chip-tool-darwin/commands/pairing/PairingDelegateBridge.h]: Done. Thank you.

File: examples/chip-tool-darwin/commands/pairing/PairingDelegateBridge.h (hunk 2 above)
Comment (krypton36) [reply examples/chip-tool-darwin/commands/pairing/PairingDelegateBridge.h]: Done. thank you.

**Comment by woody-apple** [comment]
Fast tracking given this has had > 3 days for review.

**Pull request merged**